src/DurableTask.SqlServer.AzureFunctions/SqlDurabilityProvider.cs

```diff
@@ -23,8 +28,51 @@ class SqlDurabilityProvider : DurabilityProvider
             : base("SQL Server", service, client, options.ConnectionStringName)
         {
             this.options = options;
+            this.service = service;
         }
 
         public override JObject ConfigurationJson => JObject.FromObject(this.options);
+
+        public override async Task<string?> RetrieveSerializedEntityState(EntityId entityId, JsonSerializerSettings serializierSettings)
+        {
+            string instanceId = entityId.ToString();
+            OrchestrationState? orchestrationState = await this.service.GetOrchestrationStateAsync(
+                instanceId,
+                executionId: null);
+
+            // Entity state is expected to be persisted as orchestration input.
+            string? entityMetadata = orchestrationState?.Input;
+            if (string.IsNullOrEmpty(entityMetadata))
+            {
+                return null;
+            }
+
+            // The entity state envelope is expected to be a JSON object.
+            JObject entityJson;
+            try
+            {
+                entityJson = JObject.Parse(entityMetadata);
+            }
+            catch (JsonException e)
+            {
+                throw new InvalidDataException($"Unable to read the entity data for {instanceId} because it's in an unrecognizeable format.", e);
+            }
+
+            // Entities that are deleted are expected to have { "exists": false }.
+            if (entityJson.TryGetValue("exists", out JToken existsValue) &&
+                existsValue.Type == JTokenType.Boolean &&
+                existsValue.Value<bool>() == false)
+            {
+                return null;
+            }
+
+            // The actual state comes from the { "state": "..." } string field.
+            if (!entityJson.TryGetValue("state", out JToken value))
```

Comment:
The Azure Storage provider has access to a `SchedulerState` that allows for type-safe deconstruction of this data. However, that type is internal and not available for use here, so I'm instead manually parsing the data.